contributor-docs/events-notation.md

```diff
@@ -0,0 +1,141 @@
+
+### Client-Only Events
+
+Events with non-zero client numbers (`e5+1`, `e5+2`, `e5+3`, etc.) are client-only and won't sync to the backend. They increment the client counter while keeping the same global number.
+
+```
+e5 → e5+1 → e5+2 → e5+3 → e6
+     └─ client-only ─┘
+```
+
+## Common Patterns
+
+### Sequential Events
+```
+e1 → e2 → e3
+```
+
+### Concurrent Events
+```
+e1 ║ e2
+```
+
+### Parent Relationships
+```
+e1 (parent: e0)
+e2 (parent: e1)
+e2+1 (parent: e2)
+```
+
+## Sync Scenarios
+
+### Unconfirmed Events
+
+Events that haven't been confirmed by the sync backend are denoted with a single quote:
+- `e3'` - Unconfirmed event (pending confirmation from sync backend)
+
+### Notation for Client-Specific Unconfirmed Events
+
+When multiple clients create unconfirmed events independently, we use prefix notation to distinguish them:
+- `A:e3'` - Client A's unconfirmed e3
+- `B:e3'` - Client B's unconfirmed e3
+
+### Partition and Sync Example
+
+Two clients start with shared history, partition, create local events offline, then sync:
+
+```
+Initial state (all synced):
+Client A:      e1 → e2
+Client B:      e1 → e2
+Sync Backend:  e1 → e2
+
+Network partition occurs:
+════════════════════════════════════════════════
+
+Offline activity (both clients number from e3):
+Client A:      e1 → e2 → A:e3' → A:e4'
+Client B:      e1 → e2 → B:e3' → B:e4' → B:e5'
+Sync Backend:  e1 → e2 (unchanged)
+
+Network restored, sync begins:
+────────────────────────────────────────────────
+
+Client A syncs first:
+Client A:      e1 → e2 → e3 → e4  (confirmed)
+Sync Backend:  e1 → e2 → e3 → e4
+
+Client B syncs (rebase required):
+Client B before:   e1 → e2 → B:e3' → B:e4' → B:e5'
+Client B rebased:  e1 → e2 → e3 → e4 → B:e5'(←e3') → B:e6'(←e4') → B:e7'(←e5')  (unconfirmed)
+Client B after:    e1 → e2 → e3 → e4 → e5 → e6 → e7  (confirmed)
+Sync Backend:      e1 → e2 → e3 → e4 → e5 → e6 → e7
+
+Final state (all synced):
+Client A:      e1 → e2 → e3 → e4 → e5 → e6 → e7  (after pull)
+Client B:      e1 → e2 → e3 → e4 → e5 → e6 → e7
+Sync Backend:  e1 → e2 → e3 → e4 → e5 → e6 → e7
+```
+
+Notes:
+- Each client independently numbers their unconfirmed events starting from e3'
+- The prefix `A:` or `B:` clarifies which client created each unconfirmed event
+- The sync backend assigns the final authoritative sequence numbers
+- Rebasing happens in two steps: first events are rebased locally (still unconfirmed), then confirmed by the sync backend
+- The `(←e3')` notation shows the origin: `B:e5'(←e3')` means this is B's e5' which was originally e3'
+
+## Event Context Hints
+
+Add context hints to clarify what events do:
+
+```
+Client A: e1 → e2{createUser} → e3'{updateUser.name} → e4'{deletePost.123}
+Client B: e1 → e2 → e3'{user.456} → e4'{post.create}
+
+Client B rebased: e1 → e2 → e3 → e4 → B:e5'{user.456}(←e3') → B:e6'{post.create}(←e4')
+```
+
+Context hint formats:
+- `{actionType}` - General action: `e2{createUser}`
```

Comment:
Shouldn't this be `eventName` (instead of `actionType`) and in the past tense (`userCreated` instead of `createUser`)?